docs/sop.md

```diff
@@ -1,152 +1,554 @@
+   - `ClowdApp`: Namespace-scoped resource defining application specifications
+   - `ClowdJobInvocation`: Resource for managing job executions
 
-#### Target Namespace
+3. **Provider System**
+   - Modular architecture supporting different infrastructure modes
+   - Providers: Database, Kafka, Object Storage, Logging, Metrics, Web, etc.
+   - Each provider supports multiple modes (local, operator, app-interface)
 
-Environmental resources, such as the Kafka/Zookeeper from the example in the *Modes* section, will
-be placed in the ``ClowdEnvironment``'s target namespace. This is configured by setting the
-``targetNamespace`` attribute of the ``ClowdEnvironment``. If it is omitted, a random target
-namespace is generated instead. The name of this resource can be found by inspecting the
-``status.targetNamespace`` of the ClowdEnvironment resource.
+#### Data Flow
 
-### ClowdApp
+```
+ClowdApp → Controller → Provider Logic → K8s Resources → Application Config
+    ↓           ↓              ↓              ↓              ↓
+ Spec      Reconcile     Infrastructure   Deployments   cdappconfig.json
+```
 
-**abbreviated to [app] in k8s**
+#### Key Concepts
 
-The ``ClowdApp`` CRD is responsible for configuring an application and is namespace scoped. Any
-resources Clowder creates on behalf of the application will reside in the same namespace that the
-``ClowdApp`` resources is applied to. As such the ``ClowdApp`` name must be unique within a
-particular namespace.  An ``ClowdApp`` does not have to be placed in the ``ClowdEnvironment``'s
-target namespace.
+- **Environment Coupling**: ClowdApps are coupled to ClowdEnvironments via `envName`
+- **Configuration Generation**: Apps receive standardized config via mounted secrets
+- **Dependency Management**: Automatic service discovery and configuration injection
+- **Provider Modes**: Flexible infrastructure provisioning strategies
 
-A ``ClowdApp`` may define multiple services inside it. These services, though defined by a
-specification that is very similar to the k8s pod specification, will be deployed as individual
-deployment resources.  Functionally, defining multiple applications in the same ``ClowdApp``
-specification allows the sharing of some infrastructure dependencies such as databases.
-Applications in different ClowdApp's should not expect to be able to share databases.
+### Deployment Architecture
 
-A ``ClowdApp`` is coupled to a ``ClowdEnvironment`` by the use of the ``envName`` parameter of the
-``ClowdApp``. When Clowder configures applications, it will point them to the resources that are
-defined in the coupled ``ClowdEnvironment``. As an example, if a ``ClowdApp`` requires the use of a
-Kafka topic, the application will be configured to use the kafka broker that has been configured in
-the coupled ClowdEnvironment, which could be a local, strimzi or app-interface managed Kafka
-instance.
+Clowder is deployed via Operator Lifecycle Manager (OLM) with the following components:
```

Comment:
We actually don't deploy it with OLM at this time. I think anything related to OLM can be removed from here for now. We simply deploy the manifest at https://github.com/RedHatInsights/clowder/blob/master/deploy.yml